### PR TITLE
fix build_root.project_image to use Dockerfile from pull.

### LINKED ISSUE #242: build_root.project_image.dockerfile_path is broken
build_root.project_image.dockerfile_path is taken from master instead of the branch that's being tested meaning you can never succeed adding that file. (Also means CI let's you delete that file and get permanently broken.)

```
2019/01/20 17:21:48 Resolved source https://github.com/tnozicka/openshift-acme to master@7086e885, merging: #79 a84ec983 @tnozicka
2019/01/20 17:21:48 Resolved openshift/origin-v3.11:base to sha256:82ade8718e72f34b5eac3b279aaf659e2d45eb9f24758d2e022d69ba1c5c2c71
2019/01/20 17:21:48 Using namespace ci-op-lrq17gfq
2019/01/20 17:21:48 Running root, src, bin, verify
2019/01/20 17:21:48 Creating namespace ci-op-lrq17gfq
2019/01/20 17:21:48 Creating rolebinding for user tnozicka in namespace ci-op-lrq17gfq
2019/01/20 17:21:48 Setting a soft TTL of 1h0m0s for the namespace
2019/01/20 17:21:48 Setting a hard TTL of 12h0m0s for the namespace
2019/01/20 17:21:48 Setting up pipeline imagestream for the test
2019/01/20 17:21:48 Building root
2019/01/20 17:21:58 Build root failed, printing logs:
Cloning "https://github.com/tnozicka/openshift-acme.git" ...
	Commit:	7086e885628c6d03825af18e4e62bd02810a6e3d (Add OWNERS)
	Author:	Tomas Nozicka <tnozicka@gmail.com>
	Date:	Thu Aug 30 17:20:04 2018 +0200
error: open /tmp/build/inputs/images/build/Dockerfile: no such file or directory
```

/kind bug

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/tnozicka_openshift-acme/79/pull-ci-tnozicka-openshift-acme-master-verify/46

### PR DESCRIPTION
**Pull request by droslean**
closes #242

### CODE REVIEW

File: pkg/steps/git_source.go [hunk 1]
```diff
@@ -23,12 +23,16 @@ func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefiniti
 }
 
 func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
+	ref := s.jobSpec.Refs.BaseRef
+	if len(s.jobSpec.Refs.Pulls) > 0 && s.jobSpec.Refs.Pulls[0].Number != 0 {
```
Comment (tnozicka) [inline pkg/steps/git_source.go]: what if this is the batch merge where you have multiple pulls? given it's an array

Comment (droslean) [reply pkg/steps/git_source.go]: for that reason, I get only the first value.  If the batch job will have other PRs that are changing the Dockerfile, they will be ignored.

Comment (droslean) [reply pkg/steps/git_source.go]: Do you have any other suggestions?

**Comment by stevekuznetsov** [comment]
We didn't do this as it would have required us to know the content of the Dockerfile when doing the input resolution steps for idepotency. The behavior that is "broken" is working as implemented and documented in the feature docs...

File: pkg/steps/git_source.go (hunk 1 above)
Comment (tnozicka) [reply pkg/steps/git_source.go]: clone master, merge all the PRs start build from local dir? or push the batch to some remote temporarily

**Pull request closed without merging**